File: Python3/프로그래머스/0/120884. 치킨 쿠폰/치킨 쿠폰.py
def solution(chicken):
    chickens = chicken // 10
    coupons = chicken % 10
    
    prev_chickens = chickens
    while True:
        coupons += prev_chickens
        
        chickens += coupons // 10
        
        prev_chickens = coupons // 10
        coupons = coupons % 10 
        
        if prev_chickens + coupons < 10:
            break
    
    return chickens
        

File: Swift/프로그래머스/0/120884. 치킨 쿠폰/치킨 쿠폰.swift
import Foundation

func solution(_ chicken:Int) -> Int {
    var chickens = chicken / 10
    var coupons = chicken % 10
    var prev_chickens = chickens
    
    while true {
        coupons += prev_chickens
        chickens += coupons / 10
        prev_chickens = coupons / 10
        coupons = coupons % 10
        
        if prev_chickens + coupons < 10 {
            break
        }
    }
    return chickens
}